server/manager/src/main/java/org/apache/accumulo/manager/ManagerTime.java

```diff
@@ -58,8 +62,7 @@ public ManagerTime(Manager manager, AccumuloConfiguration conf) throws IOExcepti
 
     try {
       zk.putPersistentData(zPath, "0".getBytes(UTF_8), NodeExistsPolicy.SKIP);
-      skewAmount =
-          new AtomicLong(Long.parseLong(new String(zk.getData(zPath), UTF_8)) - System.nanoTime());
+      skewAmount = new AtomicLong(updatedSkew(zk.getData(zPath)));
```

Comment:
Would skewAmount be a good candidate for conversion to [NanoTime](https://github.com/apache/accumulo/blob/main/core/src/main/java/org/apache/accumulo/core/util/time/NanoTime.java)? Not sure if it would be in the scope of this PR but thought I would point it out.

Maybe the `time` member variable in `SteadyTime` too.

Reply:
I'm not sure what you mean, this is already using nanotime. Both the skew amount and the time member variable are nanotime...this just wraps it in a stronger type. Elsewhere in the code it uses millis and does a conversion

Reply:
There is a util object that was added to accumulo called `NanoTime` [linked here](https://github.com/apache/accumulo/blob/main/core/src/main/java/org/apache/accumulo/core/util/time/NanoTime.java). It does something and creates a stronger type with safe operation of `System.nanoTime()`. It might not be the best fit here but thought I would mention it just in case.

Reply:
Ah ok got it, yeah I didn't click on the link you have for NanoTime, I should probably have done that. :)

That's interesting, that might be useful here I will take a look at it and see.

Reply:
I ended up not using NanoTime and went with Duration instead as it makes a lot more sense in this case, I will comment more in a separate comment.